src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -91,25 +92,41 @@ def ensemble(self):
         return self._ensemble
 
     async def _fm_handler(self, events):
-        snapshot_update_event = self.ensemble.update_snapshot(events)
-        await self._send_snapshot_update(snapshot_update_event)
+        with self._snapshot_mutex:
+            snapshot_update_event = self.ensemble.update_snapshot(events)
+        send_future = asyncio.run_coroutine_threadsafe(
+            self._send_snapshot_update(snapshot_update_event), self._loop
+        )
+        send_future.result()
 
     async def _started_handler(self, events):
         if self.ensemble.status != ENSEMBLE_STATE_FAILED:
-            snapshot_update_event = self.ensemble.update_snapshot(events)
-            await self._send_snapshot_update(snapshot_update_event)
+            with self._snapshot_mutex:
+                snapshot_update_event = self.ensemble.update_snapshot(events)
+            send_future = asyncio.run_coroutine_threadsafe(
+                self._send_snapshot_update(snapshot_update_event), self._loop
```

Comment:
now that we run send snapshot in the ensemble evaluator's asyncio loop, can we remove all asyncio stuff from batchy dispatchy @JHolba ?

Reply:
Jon is working on it, resolving this comment